tests/models/encoder_decoders/encoder_decoder.py

```diff
@@ -0,0 +1,12 @@
+# pylint: disable=invalid-name
+from allennlp.common.testing import ModelTestCase
+
+
+class EncoderDecoderTest(ModelTestCase):
```

Comment:
This test isn't getting run, probably because the filename doesn't contain `test`.  Also, can you add a test for the dataset reader?